src/map.h

```diff
@@ -260,12 +260,15 @@ class map
         void set_pathfinding_cache_dirty( const int zlev );
         /*@}*/
 
-        void set_memory_seen_cache_dirty( const tripoint p ) {
-            get_cache( p.z ).map_memory_seen_cache[ p.x + p.y * MAPSIZE_Y ] = false;
+        void set_memory_seen_cache_dirty( const tripoint &p ) {
+            const int offset = p.x + ( p.y * MAPSIZE_Y );
+            if( offset >= 0 && offset < MAPSIZE_X * MAPSIZE_Y ) {
```

Comment:
I'm pretty sure this logic is wrong.  If `p.x` is out of bounds then this can wrap around to set a dirty bit on the wrong row.  I guess technically it doesn't actually matter, since extra dirty bits are safe, but it seems dubious.

Reply:
I guess we can just call `generic_inbounds` function here, similar to what's called in lightmap or scent_map with boundaries of `x=0...MAPSIZE_X, y = 0...MAPSIZE_Y`.